### PR TITLE
Introduce business rules mechanism

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR contains work related to business rule mechanism:

- interface which will be implemented by each concrete business rule
- business rule validation exception
- business rule validator that validates concrete business rule

### CODE REVIEW

File: Src/Fitnet/Shared/BusinessRulesEngine/IBusinessRule.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace SuperSimpleArchitecture.Fitnet.Shared.BusinessRulesEngine;
+
+internal interface IBusinessRule
```
Comment (kamilbaczek) [inline Src/Fitnet/Shared/BusinessRulesEngine/IBusinessRule.cs:3]:
```suggestion
public interface IBusinessRule
```
BusinessRule is a well-defined interface that can be used by several modules.  I think this is a candidate for public interface. 

What do you think about that :D ?

File: Src/Fitnet.UnitTests/GlobalUsings.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Global using directives
+
+global using System.Net;
+global using FluentAssertions;
+global using Xunit;
+global using Bogus;
```
Comment (kamilbaczek) [inline Src/Fitnet.UnitTests/GlobalUsings.cs:6]: Bogus is not used yet? Should we keep or remove it?

File: Src/Fitnet.UnitTests/BusinessRulesEngine/FakeBusinessRule.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+    private readonly int _someNumber;
+    
+    internal FakeBusinessRule(int someNumber)
+    {
+        _someNumber = someNumber;
+    }
+    
+    public bool IsMet()
+    {
```
Comment (kamilbaczek) [inline Src/Fitnet.UnitTests/BusinessRulesEngine/FakeBusinessRule.cs:15]: public bool IsMet() => _someNumber > 10;

Comment (kamilbaczek) [reply Src/Fitnet.UnitTests/BusinessRulesEngine/FakeBusinessRule.cs:15]: Could we use arrow style when it is possible? What do you think?

**Review by kamilbaczek — COMMENTED** [review]
For me

File: Src/Fitnet.UnitTests/BusinessRulesEngine/FakeBusinessRule.cs (hunk 3 above)
Comment (meaboutsoftware) [reply Src/Fitnet.UnitTests/BusinessRulesEngine/FakeBusinessRule.cs:15]: We can :)

File: Src/Fitnet.UnitTests/GlobalUsings.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Src/Fitnet.UnitTests/GlobalUsings.cs:6]: You are right, I will drop it

File: Src/Fitnet/Shared/BusinessRulesEngine/IBusinessRule.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Src/Fitnet/Shared/BusinessRulesEngine/IBusinessRule.cs:3]: In the first round I would like to keep it internal (as we just have 1 project) - the goal is to have as less questionable things as possible

**Pull request merged**